compiler/luci-interpreter/src/kernels/Prelu.cpp

```diff
@@ -128,25 +149,97 @@ void Prelu::evalQuantized() const
-        input_val >= 0
-            ? tflite::MultiplyByQuantizedMultiplier(input_val, _output_multiplier_identity,
-                                                    _output_shift_identity)
-            : tflite::MultiplyByQuantizedMultiplier(input_val * alpha_val, _output_multiplier_alpha,
-                                                    _output_shift_alpha);
-    const int32_t clamped_output = std::min(quantized_max, std::max(quantized_min, output_val));
-    return static_cast<int16_t>(clamped_output);
-  };
-
-  BinaryOpBroadcastSlow(getTensorShape(input()), getTensorData<int16_t>(input()),
-                        getTensorShape(alpha()), getTensorData<int16_t>(alpha()),
-                        getTensorShape(output()), getTensorData<int16_t>(output()), fn);
+  const int32_t output_val =
+      input_val >= 0 ? tflite::MultiplyByQuantizedMultiplier(input_val, identity_mult.multiplier,
+                                                             identity_mult.shift)
+                     : tflite::MultiplyByQuantizedMultiplier(
+                           input_val * alpha_val, alpha_mult.multiplier, alpha_mult.shift);
+  const int32_t clamped_output = std::min(quantized_max, std::max(quantized_min, output_val));
+  return clamped_output;
+}
+
+void Prelu::evalQuantizedS16() const
+{
+  bool channel_wise = alpha()->scales().size() != 1;
+  if (!channel_wise)
+  {
+    auto fn = [this](int16_t input_val, int16_t alpha_val) {
+      const ChannelQuantMultipliers pos_mult{_output_shift_identity, _output_multiplier_identity};
+      const ChannelQuantMultipliers &neg_mult = _alpha_multipliers[0];
+      return evalElemS16Prelu(input_val, alpha_val, pos_mult, neg_mult);
+    };
+
+    BinaryOpBroadcastSlow(getTensorShape(input()), getTensorData<int16_t>(input()),
+                          getTensorShape(alpha()), getTensorData<int16_t>(alpha()),
+                          getTensorShape(output()), getTensorData<int16_t>(output()), fn);
+  }
+  else
+  {
+    tflite::RuntimeShape input_shape = getTensorShape(input());
+    tflite::RuntimeShape alpha_shape = getTensorShape(alpha());
+    tflite::RuntimeShape output_shape = getTensorShape(output());
+    const int16_t *input_data = input()->data<int16_t>();
+    const int16_t *alpha_data = alpha()->data<int16_t>();
+    int16_t *output_data = output()->data<int16_t>();
+
+    constexpr int N = 5;
+
+    const ChannelQuantMultipliers pos_mult{_output_shift_identity, _output_multiplier_identity};
+
+    if (input_shape == alpha_shape)
+    {
+      const int flat_size = tflite::MatchingElementsSize(input_shape, alpha_shape, output_shape);
+      int quantized_dim = alpha()->quantized_dimension();
+
+      int32_t outer_dims_size = 1;
+      for (int i = 0; i < quantized_dim; ++i)
+        outer_dims_size *= input_shape.Dims(i);
+      int32_t quant_dim_size = alpha_shape.Dims(alpha()->quantized_dimension());
+      int32_t inner_dims_size = flat_size / outer_dims_size / quant_dim_size;
```

Comment:
It seems that this code assumes `quantized_dimension` can be any integer in [0, rank()-1], but there is a restriction in our quantization spec.

In CWQ (**channel-wise** quantization), `quantized_dimension` of alpha must be the channel dimension, which is the last dimension of the tensor (= rank - 1). And the length of `quantized_dimension` must be the same with that of the last dimension of input (=channel size). 

For example, let's assume input shape is [1,4,4,64]. Here, alpha shape is something like [1,1, 64], and the `quantized_dimension` is 2 (rank - 1). If alpha shape is [64], then the quantized_dimension will be 0 (rank - 1).

So, `inner_dims_size` in this code is always 1.